impl/pkg/service/pkarr.go

```diff
@@ -123,15 +136,24 @@ func fromPKARRRecord(record storage.PKARRRecord) (*GetPKARRResponse, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &GetPKARRResponse{
+	return &GetPkarrResponse{
 		V:   vBytes,
 		Seq: record.Seq,
 		Sig: [64]byte(sigBytes),
 	}, nil
 }
 
-// GetPKARR returns the full PKARR (including sig data) for the given z-base-32 encoded ID
-func (s *PKARRService) GetPKARR(ctx context.Context, id string) (*GetPKARRResponse, error) {
+// GetPkarr returns the full Pkarr record (including sig data) for the given z-base-32 encoded ID
+func (s *PkarrService) GetPkarr(ctx context.Context, id string) (*GetPkarrResponse, error) {
+	// first do a cache lookup
+	if s.cache.Has(id) {
+		cacheItem := s.cache.Get(id)
+		until := cacheItem.ExpiresAt().Sub(time.Now())
+		logrus.Debugf("resolved pkarr record<%s> from cache, with remaining TTL: %s", id, until)
+		return fromPkarrRecord(cacheItem.Value())
+	}
```

Comment:
Curious why you chose this pattern vs using a loader? It simplifies access so pkarr so that everything goes through the cache, instead of selectively going through it.

Reply:
Loader won't work since it needs to be
1. Check cache
2. DHT
3. Fallback to storage

Reply:
Unless I put this logic in the loader but what's the difference?

Reply:
Very small difference. With the loader pattern, then other places in the program that make a call to `s.cache.Get()` will not have to worry about it... Not a big deal. 

On a separate note, is it even worth thinking about how to protect against running out of memory? The cache can grow unbounded at this point.